### PR TITLE
[COARSE] Add micro benchmark

### PR DESCRIPTION
**Pull request by ldorau**

### CODE REVIEW

**Comment by bratpiorka** [comment]
could you make this PR a draft and add "WIP" or something? I think it is not ready to merge since we don't want to merge coarse provider yet

File: benchmark/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Copyright (C) 2023 Intel Corporation
+# Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+if(CMAKE_BUILD_TYPE STREQUAL "Debug")
+	add_link_options(-O0)
+	if(WIN32)
```
[the remark this answers is not in the document]
Comment (ldorau) [reply benchmark/CMakeLists.txt:7]: Done

**Comment by ldorau** [comment]
> could you make this PR a draft and add "WIP" or something? I think it is not ready to merge since we don't want to merge coarse provider yet

But it is a PR to the [oneapi-src:coarse_provider](https://github.com/oneapi-src/unified-memory-framework/tree/coarse_provider) branch.

File: src/pool/pool_disjoint.cpp [hunk 2]
```diff
@@ -433,16 +433,11 @@ Slab::~Slab() {
     try {
         memoryProviderFree(bucket.getMemHandle(), MemPtr);
     } catch (MemoryProviderError &e) {
```
[the remark this answers is not in the document]
Comment (ldorau) [reply src/pool/pool_disjoint.cpp]: It was a mistake - reverted.

[the remark this answers is not in the document]
Comment (ldorau) [reply src/pool/pool_disjoint.cpp]: BTW see https://github.com/oneapi-src/unified-memory-framework/issues/63

**Comment by ldorau** [comment]
@bratpiorka PR updated

**Pull request merged**